sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,692 @@
+
+    private static final ResponseConstructorsCache RESPONSE_CONSTRUCTORS_CACHE = new ResponseConstructorsCache();
+
+    // RestProxy is a commonly used class, use a static logger.
+    private static final ClientLogger LOGGER = new ClientLogger(SyncRestProxy.class);
+
+    private final HttpPipeline httpPipeline;
+    private final SerializerAdapter serializer;
+    private final SwaggerInterfaceParser interfaceParser;
+    private final HttpResponseDecoder decoder;
+
+    /**
+     * Create a RestProxy.
+     *
+     * @param httpPipeline the HttpPipelinePolicy and HttpClient httpPipeline that will be used to send HTTP requests.
+     * @param serializer the serializer that will be used to convert response bodies to POJOs.
+     * @param interfaceParser the parser that contains information about the interface describing REST API methods that
+     * this RestProxy "implements".
+     */
+    private SyncRestProxy(HttpPipeline httpPipeline, SerializerAdapter serializer, SwaggerInterfaceParser interfaceParser) {
+        this.httpPipeline = httpPipeline;
+        this.serializer = serializer;
+        this.interfaceParser = interfaceParser;
+        this.decoder = new HttpResponseDecoder(this.serializer);
+    }
+
+    /**
+     * Get the SwaggerMethodParser for the provided method. The Method must exist on the Swagger interface that this
+     * RestProxy was created to "implement".
+     *
+     * @param method the method to get a SwaggerMethodParser for
+     * @return the SwaggerMethodParser for the provided method
+     */
+    private SwaggerMethodParser getMethodParser(Method method) {
+        return interfaceParser.getMethodParser(method);
+    }
+
+    /**
+     * Send the provided request asynchronously, applying any request policies provided to the HttpClient instance.
+     *
+     * @param request the HTTP request to send
+     * @param contextData the context
+     * @return a {@link Mono} that emits HttpResponse asynchronously
+     */
+    public HttpResponse send(HttpRequest request, Context contextData) {
+        return httpPipeline.sendSynchronously(request, contextData);
+    }
+
+    @Override
+    public Object invoke(Object proxy, final Method method, Object[] args) {
+        validateResumeOperationIsNotPresent(method);
+
+        try {
+            final SwaggerMethodParser methodParser = getMethodParser(method);
+            final HttpRequest request = createHttpRequest(methodParser, args);
+            Context context = methodParser.setContext(args);
+
+            RequestOptions options = methodParser.setRequestOptions(args);
+            context = mergeRequestOptionsContext(context, options);
+
+            context = context.addData("caller-method", methodParser.getFullyQualifiedMethodName())
+                .addData("azure-eagerly-read-response", shouldEagerlyReadResponse(methodParser.getReturnType()));
+            context = startTracingSpan(method, context);
```

Comment:
we should probably add generic `catch (Exception)` and `finally` to `try` block and make sure to close the tracing span.